### DISCUSSION TITLE
Runtime error when simulating a large amount of agents.

### DISCUSSION THREAD

**Discussion by mohitmasiwal95** [PedestrianDynamics/jupedsim#1427] [Q&A]
Hi, I am trying to simulate an airport scenario using Jupedsim, and when I am trying to simulate a scenario with 200 agents, 4 different entry points, and some facilities and obstacles in the environment with new agents being added at an interval of 500 iteration counts, I am getting a runtime error, which I think tells us that the agent's model, I am using collision-free speed model V2, calculates the next position of the agent to be in an area that is inaccessible, but when I simulate with fewer agents around 50, I am not getting the runtime error. Is this something related to the agent's model or is it related to my code?

PS: I am using a json file to load the environment and also using csv writer to write the agent's spatial data at every iteration count, my model's dt = 0.1.

Thank you.

**Comment by chraibi** [comment]
Since you did not show any code, from your description I guess, that the short answer to your question is: The error is related to your code.

There are many reasons why your agents, who are initially in the simulation area, get “kicked” out and hence you get this error "agents outside the walkable are"

One of these reasons may be the very high dt you are using. For some background why this is the case, check this [explanation](https://pedestriandynamics.org/models/collision_free_speed_model/#balancing-collision-avoidance-with-performance-selecting-the-appropriate-time-step).

To debug your code, a good start would be to take `dt=0.001` and check if you still get errors. 
You can post some screenshots of the errors or snippets of your code.

One other reason could be that there is plenty of space for 50 agents to be distributed, but not so much for 200!

**Comment by Ozaq** [comment]
Hello @awtishtikh I think the issue here is the chosen dT of 0.1s. Generally the models are calibrated to the default parameters to work with a dT of 0.01, i.e. an update every 10ms.

With larger time steps the "collision free" property of the model breaks down and agents may intersect or leave the simulation area.

We discussed a few months back how far we could increase the time steps without negatively impacting the simulation properties. Part of this discussion and the parameters an be seen here: https://github.com/PedestrianDynamics/jupedsim/discussions/1306

Retry your simulation with the default parameters, i.e. `dT=0.01`.
